### PR TITLE
refactor: model plugin to follow new specs

### PR DESCRIPTION
**Pull request by namchuai**
Fixes: #730 

- [x] Creating new chat with Jan assistant
Screenshot 2023-11-24 at 16 40 44

- [x] Update the file structure as https://pre-release.jan.ai/specs/models#folder-structure
Screenshot 2023-11-24 at 16 39 13

- [x] Storing messages as separate file `messages.jsonl`
Screenshot 2023-11-24 at 16 41 46

### CODE REVIEW

File: web/screens/ExploreModels/ModelVersionList/index.tsx [hunk 1]
```diff
@@ -1,15 +1,13 @@
-import { ModelCatalog, ModelVersion } from '@janhq/core/lib/types'
+import { Model } from '@janhq/core/lib/types'
 
 import ModelVersionItem from '../ModelVersionItem'
 
 type Props = {
-  model: ModelCatalog
-  versions: ModelVersion[]
+  versions: Model[]
```
Comment (louis-jan) [inline web/screens/ExploreModels/ModelVersionList/index.tsx:6]: nice, no more versions. Let's rename the props as well

File: web/utils/dummy.ts [hunk 2]
```diff
@@ -14,37 +14,121 @@ export const dummyModel: ModelCatalog = {
...
+      version: '1.0.0',
+      source_url:
         'https://huggingface.co/aladar/TinyLLama-v0-GGUF/resolve/main/TinyLLama-v0.Q8_0.gguf',
+      id: 'TinyLLama-v0.Q8_0.gguf',
+      name: 'TinyLLama-v0.Q8_0.gguf',
+      owned_by: 'you',
+      created: 0,
+      description: '',
+      state: ModelState.ToDownload,
+      parameters: {
```
Comment (louis-jan) [inline web/utils/dummy.ts:27]: lol now we duplicate metadata (model manifest) for each version

File: web/screens/ExploreModels/ModelVersionItem/index.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import React, { useMemo } from 'react'
 
-import { ModelCatalog, ModelVersion } from '@janhq/core/lib/types'
-import { Button, Badge } from '@janhq/uikit'
-
+import { Model } from '@janhq/core/lib/types'
+import { Button } from '@janhq/uikit'
+import { Badge } from '@janhq/uikit'
```
Comment (louis-jan) [inline web/screens/ExploreModels/ModelVersionItem/index.tsx:6]:
```suggestion
import { Button, Badge } from '@janhq/uikit'
```

**Comment by louis-jan** [comment]
Love it! Great work ⚡️ @namchuai 🚀

File: web/utils/dummy.ts (hunk 2 above)
Comment (namchuai) [reply web/utils/dummy.ts:27]: yeah, it's for current UI implementation. In the future, I think we might remove it

File: core/src/plugins/inference.ts [hunk 4]
```diff
@@ -7,9 +7,9 @@ import { JanPlugin } from "../plugin";
 export abstract class InferencePlugin extends JanPlugin {
   /**
    * Initializes the model for the plugin.
-   * @param modelFileName - The name of the file containing the model.
+   * @param modelId - The ID of the model to initialize.
    */
-  abstract initModel(modelFileName: string): Promise<void>;
+  abstract initModel(modelId: string): Promise<void>;
```
Comment (louis-jan) [inline core/src/plugins/inference.ts:12]: I think we will put params to this

File: core/src/plugins/assistant.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+ * Abstract class for assistant plugins.
+ * @extends JanPlugin
+ */
+export abstract class AssistantPlugin extends JanPlugin {
+  abstract createAssistant(assistant: Assistant): Promise<void>;
+
+  abstract deleteAssistant(assistant: Assistant): Promise<void>;
+
+  abstract getAssistants(): Promise<Assistant[]>;
+}
```
Comment (louis-jan) [inline core/src/plugins/assistant.ts:14]:
```suggestion
export abstract class AssistantPlugin extends JanPlugin {
  abstract createAssistant(assistant: Assistant): Promise<void>;
 
  abstract updateAssistant(assistant: Assistant): Promise<void>;

  abstract deleteAssistant(assistant: Assistant): Promise<void>;

  abstract getAssistants(): Promise<Assistant[]>;
}
```

File: core/src/types/index.ts [hunk 6]
```diff
@@ -98,39 +98,106 @@ export interface Thread {
  * @stored
  */
 export interface Model {
-  /** Combination of owner and model name.*/
+  /**
+   * The type of the object.
+   * Default: "model"
+   */
+  object: string;
```
Comment (louis-jan) [inline core/src/types/index.ts:105]: I think this is API's DTO level, not the from origin manifest

**Comment by louis-jan** [comment]
cc @urmauur, we will need your help on the thread settings (right panel)

**Comment by louis-jan** [comment]
NOTES: there will be a remote model settings (API_KEY...) cc @vuonghoainam

**Review by louis-jan — APPROVED** [review]
🚀

**Pull request merged**